kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -37,13 +39,44 @@ grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
     ctrl.selectedCollections = selectedCollections;
 }]);
 
+grCollectionsPanel.factory('collectionsTreeState', ['$window', function($window) {
+    // TODO: Add garbage collection to state.
+    const localStorageKey = 'collectionsTreeOpen';
+    const jsonStr = $window.localStorage.getItem(localStorageKey) || '[]';
+    const jsonArr = JSON.parse(jsonStr);
+    // A little bit of security incase this was set weirdly before.
```

Comment:
If you want to be paranoid, you could surround the `JSON.parse` with `try-catch`. Up to you.